### PR TITLE
[travis] Checking repo status after build

### PR DESCRIPTION
**Pull request by marechal-p**
#2138

### CODE REVIEW

**Comment by marechal-p** [comment]
The install command I wrote seems to have done the trick.

Thing to know: it is currently checking for any change in the repository.
We could make it look just for `yarn.lock`. Is it good enough as it is ?

**Comment by kittaakos** [comment]
> The install command I wrote seems to have done the trick.

Awesome!

> Thing to know: it is currently checking for any change in the repository.
> We could make it look just for yarn.lock. Is it good enough as it is ?

I would say it should be fine to check everything and it would be an overkill to parse the Git status and extract the file name. Perhaps we could improve the message:
```
The command "yarn && if [ $(git status -s | wc -c) -gt 0 ]; then sh -c "exit 1"; fi" failed and exited with 1 during .
```

What do you think @svenefftinge,  @marcdumais-work, and @akosyakov?

File: scripts/check_git_status.sh [hunk 1]
```diff
@@ -0,0 +1,6 @@
+#!/bin/sh
+if [ $(git status -s | wc -c) -gt 0 ];
+then
+    echo "The git repository state changed after the build, this should not happen."
```
Comment (kittaakos) [inline scripts/check_git_status.sh]: Let's print the output of `git status` too, please.

Comment (marechal-p) [reply scripts/check_git_status.sh]: Very good idea

**Comment by marcdumais-work** [comment]
> We could make it look just for yarn.lock. Is it good enough as it is ?

Why not; it's easy to grep for something more specific, that would not catch some other random unexpected changed file:
```if [ $(git status -s | grep yarn.lock | wc -c) -gt 0 ];```

We could also give a more specific error message, saying the build is failing because an up-to-date version of yarn.lock is not part of the PR, to correct and re-submit, or something.

**Comment by svenefftinge** [comment]
LGTM
I expect git status to except a path but wouldn't it be good to get notified about other changes, too? 
For instance, we might have broken the `.gitignore` such that generated files are showing up in git or some auto-formatter doing things.

**Comment by kittaakos** [comment]
> that would not catch some other random unexpected changed file:

I think it should catch every random unexpected change. I would not do additional grepping.

> give a more specific error message, saying the build is failing because an up-to-date version of yarn.lock

Yes. I would add, did you commit your `yarn.lock`? Check your `.gitignore`...

> I expect git status to except a path but wouldn't it be good to get notified about other changes, too?

What path do you mean? It runs `git status` for the entire repository and with the current logic, it will ~fail~ notify us of any other changes, not just the `yarn.lock`.

**Comment by marechal-p** [comment]
> For instance, we might have broken the .gitignore such that [...]

@svenefftinge then if it is broken and triggers the git status check this is good right ? The problem could be ignoring files with changes so that we don't detect the changes. But someone could also change the `travis.yml` file to remove the check. Idk how paranoid we should get here.

**Comment by marcdumais-work** [comment]
> Idk how paranoid we should get here.

The main immediate goal is detecting un-updated yarn.lock, so let's cover what was discussed above for now:   run ```git status``` and fail if there is any change, and in the error message suggest possible likely causes, as suggested by @kittaakos above: "did you commit your yarn.lock? Check your .gitignore..."

Sounds good?

**Comment by kittaakos** [comment]
> [...]run git status and fail if there is any change[...]
> [...]"did you commit your yarn.lock? Check your .gitignore..."[...]
> [...]Sounds good?

+1

**Comment by marechal-p** [comment]
@kittaakos @marcdumais-work tell me if this is good for you, then I'll revert the commit that fails the CI build and merge if everything is ok.

**Comment by kittaakos** [comment]
👍 It looks great, please revert and squash.

**Comment by marechal-p** [comment]
@kittaakos done.

**Comment by marcdumais-work** [comment]
@marechal-p ping

**Pull request merged**